.github/workflows/api-gateway.yml

```diff
@@ -0,0 +1,219 @@
+name: CI/CD for API Gateway
```

Comment:
What this does is the following:

1. Build and push an image for the API gateway, PostgreSQL database, and applications microservice
2. Pull down new versions of these images in the EC2 instance
3. Run `docker-compose up -f docker-compose.yml -f docker-compose.prod.yml -d` in the EC2 instance to start up all of the containers at once